File: app/src/main/java/art/luaj/hyperisle/ui/fragment/mainnav/MainNavFragment.java
package art.luaj.hyperisle.ui.fragment.mainnav;

import android.view.LayoutInflater;
import android.view.ViewGroup;

import androidx.navigation.NavController;
import androidx.navigation.fragment.NavHostFragment;
import androidx.navigation.ui.NavigationUI;

import com.google.android.material.bottomnavigation.BottomNavigationView;

import art.luaj.hyperisle.R;
import art.luaj.hyperisle.databinding.FragmentMainNavBinding;
import art.luaj.hyperisle.ext.BaseFragment;

public class MainNavFragment extends BaseFragment<FragmentMainNavBinding> {
    @Override
    protected FragmentMainNavBinding inflate(LayoutInflater inflater, ViewGroup container, boolean attachToRoot) {
        return FragmentMainNavBinding.inflate(inflater, container, attachToRoot);
    }

    @Override
    protected void initFragment(FragmentMainNavBinding binding) {
        NavHostFragment navFragment = (NavHostFragment) getChildFragmentManager().findFragmentById(R.id.mainNavView);
        NavController navController = navFragment.getNavController();
        BottomNavigationView navView = binding.mainNavBottomNav;
        navView.setLabelVisibilityMode(BottomNavigationView.LABEL_VISIBILITY_SELECTED);
        NavigationUI.setupWithNavController(navView, navController);
    }
}

File: app/src/main/java/art/luaj/hyperisle/ui/fragment/MainNavFragment.kt
package art.luaj.hyperisle.ui.fragment

import androidx.lifecycle.ViewModel
import androidx.navigation.NavOptions
import androidx.navigation.fragment.NavHostFragment
import androidx.navigation.fragment.NavHostFragment.Companion.findNavController
import androidx.navigation.ui.setupWithNavController
import art.luaj.hyperisle.R
import art.luaj.hyperisle.databinding.FragmentMainNavBinding
import art.luaj.hyperisle.ext.Tools
import art.luaj.hyperisle.ui.base.BaseFragment
import com.google.android.material.bottomnavigation.BottomNavigationView

class MainNavFragment : BaseFragment<FragmentMainNavBinding, ViewModel>(
    FragmentMainNavBinding::inflate,
    null
) {
    override fun initFragment(binding: FragmentMainNavBinding, viewModel: ViewModel?) {
        Tools.showAlert(requireContext(), "Hello World", true)
        (childFragmentManager.findFragmentById(R.id.mainNavView) as NavHostFragment).apply {
            binding.mainNavBottomNav.labelVisibilityMode =
                BottomNavigationView.LABEL_VISIBILITY_SELECTED
            binding.mainNavBottomNav.setupWithNavController(navController)
        }
    }
}